### PR TITLE
Enh/neuropixels rig

### PR DESCRIPTION
**Pull request by mochic**
Adds neuropixels rig mapping for mpe sync config [sync.yml](https://github.com/mochic/aind-metadata-mapper/blob/fc78013be5bf6c92bc1b190f4b66bc61e4ac5a8a/tests/resources/neuropixels/sync.yml) and mvr config [mvr.ini](https://github.com/mochic/aind-metadata-mapper/blob/enh/neuropixels-rig/tests/resources/neuropixels/mvr.ini). An additional non-mpe config [mvr.mapping.json](https://github.com/mochic/aind-metadata-mapper/blob/enh/neuropixels-rig/tests/resources/neuropixels/mvr.mapping.json) is used to map mvr camera names to camera assembly names in the rig. Aside from sync and camera assembly information, the rest of the rig information is static and rehydrated from [rig.partial.json](https://github.com/mochic/aind-metadata-mapper/blob/enh/neuropixels-rig/tests/resources/neuropixels/rig.partial.json).

### CODE REVIEW

File: src/aind_metadata_mapper/neuropixels/rig.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+import configparser
+from aind_data_schema import rig
+
+from ..core import BaseEtl
+
+
+class NeuropixelsRigException(Exception):
+    """General error for MVR."""
+
+
```
Comment (dyf) [inline src/aind_metadata_mapper/neuropixels/rig.py:16]: Go ahead and make pydantic models for these.

File: src/aind_metadata_mapper/neuropixels/rig.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                    "pixel_height": height,
+                    "pixel_width": width,
+                    "size_unit": "pixel",
+                    "max_frame_rate": frame_rate,
+                }
+            })
+
+        return rig.Rig.parse_obj({
+            **partial,
+            "modification_date": datetime.date.today(),
```
Comment (dyf) [inline src/aind_metadata_mapper/neuropixels/rig.py:191]: add `modification_date` to the arg list. you can mention in the defaults that `None` means `today()`.

File: src/aind_metadata_mapper/neuropixels/rig.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        Parameters
+        ----------
+        input_source : Path
+          Can be a string or a Path
+        output_directory : Path
+          The directory where to save the json files.
+        """
+        super().__init__(input_source, output_directory)
+
+    def _load_resource(self, path: pathlib.Path) -> str:
```
Comment (dyf) [inline src/aind_metadata_mapper/neuropixels/rig.py:46]: I don't think you need this function - `read_text` should say something to this effect if the file does not exist.

File: src/aind_metadata_mapper/neuropixels/rig.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+
+class NeuropixelsRigEtl(BaseEtl):
+    """Neuropixels rig ETL class. Extracts information from rig-related files
+    and transforms them into an aind-data-schema rig.Rig instance.
+    """
+
+    def __init__(
+        self,
+        input_source: pathlib.Path,
```
Comment (dyf) [inline src/aind_metadata_mapper/neuropixels/rig.py:32]: source is always a directory, right? if so you can call this `input_directory`

File: src/aind_metadata_mapper/neuropixels/rig.py (hunk 1 above)
Comment (dyf) [reply src/aind_metadata_mapper/neuropixels/rig.py:16]: e.g. 
```python
class MvrCamera(BaseModel):
    assembly_name: str
    serial_number: str
    height: int
    width: int
    frame_rate: float
```

File: pyproject.toml [hunk 5]
```diff
@@ -17,9 +17,10 @@ readme = "README.md"
 dynamic = ["version"]
 
 dependencies = [
-    "aind-data-schema==0.15.9",
+    "aind-data-schema==0.17.3",
     "aind-metadata-service[client]",
-    "scanimage-tiff-reader==1.4.1.4"
+    "scanimage-tiff-reader==1.4.1.4",
+    "pyyaml==6.0.1"
```
Comment (dyf) [inline pyproject.toml:23]: does this need to be pinned?

**Review by dyf — CHANGES_REQUESTED** [review]

**Comment by dyf** [comment]
@mochic let's please also make the name of the module more specific.  maybe: `neuropixels-mvr`.

File: pyproject.toml (hunk 5 above)
Comment (mochic) [reply pyproject.toml:23]: Maybe? Pyyaml released breaking changes between 5.0 and 6.0 so I figured pinning was safest in case of pyyaml 7.0.

**Comment by mochic** [comment]
@dyf This has beeb moved to #28

**Pull request closed without merging**